### PR TITLE
properly display errors

### LINKED ISSUE #127: [BUG] [CRITICAL] Infinite loop for Display in GooseError and GooseTaskError
See comment here:

https://github.com/tag1consulting/goose/pull/113/files#r462282759

### PR DESCRIPTION
**Pull request by jeremyandrews**
Display an actual/useful error.

Fix #127

### CODE REVIEW

File: src/goose.rs [hunk 1]
```diff
@@ -356,11 +356,26 @@ pub enum GooseTaskError {
     /// is available in `.method`.
     InvalidMethod { method: Method },
 }
+impl GooseTaskError {
+    fn describe(&self) -> &str {
+        match *self {
+            GooseTaskError::Reqwest(_) => "reqwest::Error",
+            GooseTaskError::Url(_) => "url::ParseError",
```
Comment (LionsAd) [inline src/goose.rs:363]: This is okay, but I would rather that we pass through information from child errors still, e.g. with `:`.

File: src/goose.rs [hunk 2]
```diff
@@ -356,11 +356,26 @@ pub enum GooseTaskError {
...
+            GooseTaskError::InvalidMethod { .. } => "Unrecognized HTTP request method",
+        }
+    }
+}
 
 // Define how to display errors.
 impl fmt::Display for GooseTaskError {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        fmt::Display::fmt(&self, f)
+        write!(f, "GooseTaskError: {}", self.describe())
```
Comment (LionsAd) [inline src/goose.rs:378]: What about we do:

```
if Some(err) = self.source() {
  err.fmt(f);
}
```

File: src/goose.rs (hunk 1 above)
Comment (jeremyandrews) [reply src/goose.rs:363]: the proposed implementation didn't work, I tried an alternative

File: src/lib.rs [hunk 3]
```diff
@@ -412,11 +412,34 @@ pub enum GooseError {
...
-        fmt::Display::fmt(&self, f)
+        match *self {
+            GooseError::Io(ref source) => write!(f, "GooseError: {} ({})", self.describe(), source),
+            GooseError::Reqwest(ref source) => {
+                write!(f, "GooseError: {} ({})", self.describe(), source)
+            }
+            GooseError::InvalidHost {
+                ref parse_error, ..
+            } => write!(f, "GooseError: {} ({})", self.describe(), parse_error),
+            _ => write!(f, "GooseError: {}", self.describe()),
```
Comment (LionsAd) [inline src/lib.rs:441]: It's a bummer all this double-work is needed, but it's good enough for now. We can always simplify later.

Comment (jeremyandrews) [reply src/lib.rs:441]: :+1:

**Pull request merged**